Peer code review Submission Checklist (#42)
worldbank/dime-standards: dime-coding-standards/checklists/tex/reviewer_feedback_checklist.tex

        
		\section*{Overview - First Impression}

>         Once you have received a package from your code review partner, please refer to the following list for basic recommended practices:

Reviewer: ```suggestion
        Once you have received a package from your code review partner, please check for the following general recommended practices:
```

			\item[] \dimeCheckBox{Each script is fully modular - that is to say that the scripts can be run independently from the master script, and do not depend on having the results of other scripts in memory.}
			\item[] \dimeCheckBox{All scripts are well-commented}
                \item[] \dimeCheckBox{All scripts are well-organized and formatted, such that one can easily identify functional chunks of code and evaluate whether they correctly implement the econometric or statistical process described}
			\item[] \dimeCheckBox{Tables are output as .csv or .tex files or other raw text files \textbf{ONLY}}
		\end{itemize}
	
		\vspace{.6em}

        \section*{Reproducibility (applicable only if data was shared with you)}

        \textit{\textbf{Next step:}} If you received data as part of the peer review package, attempt to run the code files. Ideally you will be able to do so by changing only the top-level directory in the master script. Take note of any reproducibility issues you observe as you will need to record them in \href{https://survey.wb.surveycto.com/collect/code_review_summary?caseid=}{this review summary form.}

        \begin{itemize}
			\setlength\itemsep{-0.1em}

> 			\item[] \dimeCheckBox{Was able to run all code files provided in the review package}

Reviewer: ```suggestion
			\item[] \dimeCheckBox{I was able to run all code files provided in the review package}
```

>                 \item[] \dimeCheckBox{Code did not run initially, but was able to identify the reason}

Reviewer: ```suggestion
                \item[] \dimeCheckBox{Code did not run initially, but I was able to identify the reason}
```

> 			\item[] \dimeCheckBox{Outputs shared (if any) replicated perfectly on the first run}

Reviewer: ```suggestion
			\item[] \dimeCheckBox{Outputs shared (if any) reproduced exactly}
```

            \item[] \dimeCheckBox{Outputs did not reproduce but I was able to determine the reason}
	
			
		\end{itemize}
		
        \vspace{.6em}

       \section*{Ease of Use and Understanding}

       \textit{\textbf{Now}}, assess how easy it is to understand the code you are reviewing. Consider whether the documentation provided is sufficient, and whether the code is organized in such a way that it would be easy for someone else to work on it. Make note of what additional documentation is needed as you will be asked to share detailed feedback in \href{https://survey.wb.surveycto.com/collect/code_review_summary?caseid=}{this review summary form.}

       
        \begin{itemize}
			\setlength\itemsep{-0.1em}

> 			\item[] \dimeCheckBox{Understood the code from original documentation and README file}

Reviewer: ```suggestion
			\item[] \dimeCheckBox{I understood the code from original documentation and README file}
```

                \item[] \dimeCheckBox{The code is organized such that if I had to change a core task I could do it by making changes in only one place}

                \item[] \dimeCheckBox{A new RA joining the project would be able to understand and contribute to the code}
                
                \dimeTextField{time}{1 cm}{How many days are needed to understand the code well enough to contribute? \textit{\textbf{(0-3 days)}}}
               
			
		\end{itemize}

        \vspace{.6em}

        \section*{Coding Practices}

>         \textit{\textbf{Next}}, select the box next to each pracitce if the mentioned coding practice was used in some form in the code package. Also make note of whether it was implemented perfectly, or if there is room for improvement. You will need to provide this information in \href{https://survey.wb.surveycto.com/collect/code_review_summary?caseid=}{this detailed review summary form.}

Reviewer: ```suggestion
        \textit{\textbf{Next}}, select the box next to each practice if the mentioned coding practice was used in some form in the code package. Also make note of whether it was implemented perfectly, or if there is room for improvement. You will need to provide this information in \href{https://survey.wb.surveycto.com/collect/code_review_summary?caseid=}{this detailed review summary form.}
```

        \vspace{.6em}

        \section*{Final Comments}
        \textit{\textbf{Finally,}}  make note of the main strengths of the code, suggestions on how the coder can improve, and any additional feedback from your side. You will need to provide this information in \href{https://survey.wb.surveycto.com/collect/code_review_summary?caseid=}{this detailed review summary form.)}
          

>           \dimeTextField{strengths}{8 cm}{\textbf{Three main strengths of the code reviewed:}}

Reviewer: ideally would make three boxes

Reviewer: or, just get rid of the boxes and space these lines out more
